src/sas/system/config/config_meta.py

```diff
@@ -205,17 +205,37 @@ def _generate_schema(self) -> Dict[str, SchemaElement]:
         return schema
 
     def __setattr__(self, key, value):
-        if hasattr(self, "_locked") and self._locked:
-            if key not in self.__dict__:
-                raise ConfigLocked("New attribute attempt")
 
+        # This section deals with control variables for the config
+
+        if not hasattr(self, "_meta_attributes"):
+
+            # The class is not set up at this point, don't do any checks
+            super().__setattr__(key, value)
+            return
+
+        else:
+
+            # The class is set up, check whether we are trying to set a control variable
+            if key in self._meta_attributes:
+                super().__setattr__(key, value)
+                return
+
+            # otherwise continue to part that handles values
+
+        # This section deals with config values themselves
         if getattr(self, "_locked", False):
             try:
                 super().__setattr__(key, self._schema[key].coerce(value))
             except CoercionError:
                 raise TypeError(f"Tried to set bad value '{value}' to config entry of type '{self._schema[key]}'")
         else:
+
+            if key not in self.__dict__:
```

Comment:
The config isn't locked at this point, but you're throwing a ConfigLocked exception if the key isn't already set.